codec/src/main/java/io/netty/handler/codec/ByteString.java

```diff
@@ -0,0 +1,769 @@
+        return result;
+    }
+
+    public char parseChar() {
+        return parseChar(0, length());
+    }
+
+    public char parseChar(int start, int end) {
+        return toString(start, end).charAt(0);
+    }
+
+    public short parseShort() {
+        return parseShort(0, length(), 10);
+    }
+
+    public short parseShort(int radix) {
+        return parseShort(0, length(), radix);
+    }
+
+    public short parseShort(int start, int end) {
+        return parseShort(start, end, 10);
+    }
+
+    public short parseShort(int start, int end, int radix) {
+        int intValue = parseInt(start, end, radix);
+        short result = (short) intValue;
+        if (result != intValue) {
+            throw new NumberFormatException(subSequence(start, end).toString());
+        }
+        return result;
+    }
+
+    public float parseFloat() {
+        return parseFloat(0, length());
+    }
+
+    public float parseFloat(int start, int end) {
+        return Float.parseFloat(toString(start, end));
+    }
+
+    public double parseDouble() {
+        return parseDouble(0, length());
+    }
+
+    public double parseDouble(int start, int end) {
+        return Double.parseDouble(toString(start, end));
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof ByteString)) {
+            return false;
+        }
+
+        if (this == obj) {
+            return true;
+        }
+
+        ByteString that = (ByteString) obj;
+        if (length() != that.length() || hashCode() != that.hashCode()) {
+            return false;
+        }
+
+        byte[] thisValue = value;
+        byte[] thatValue = that.value;
+        int end = thisValue.length;
+        for (int i = 0, j = 0; i < end; i++, j++) {
+            if (thisValue[i] != thatValue[j]) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Translates the entire byte string to a {@link String}.
+     * @see {@link #toString(int, int)}
+     */
+    @Override
+    public String toString() {
+        return toString(0, length());
+    }
+
+    /**
+     * Translates the entire byte string to a {@link String} using the {@code charset} encoding.
+     * @see {@link #toString(Charset, int, int)}
+     */
+    public String toString(Charset charset) {
+        return toString(charset, 0, length());
+    }
+
+    /**
+     * Translates the [{@code start}, {@code end}) range of this byte string to a {@link String}.
+     * @see {@link #toString(Charset, int, int)}
+     */
+    public String toString(int start, int end) {
+        return toString(CharsetUtil.ISO_8859_1, start, end);
+    }
+
+    /**
+     * Translates the [{@code start}, {@code end}) range of this byte string to a {@link String}
+     * using the {@code charset} encoding.
+     */
+    public String toString(Charset charset, int start, int end) {
+        int length = end - start;
+        if (length == 0) {
+            return "";
```

Comment:
This is private still.....thought I remember commenting on this for one of your PRs and you made it public?  O well I'll also make it public...